doc/pxe.md

```diff
@@ -0,0 +1,111 @@
+# Using PXE and Iguana
+
+This document explains how to run D-Installer on PXE with the help of Iguana. The described setup
+uses libvirt, but you can adapt the overall approach to other scenarios (like running your own
+TFTP server).
+
+The process can be summarized in these steps:
+
+1. Set up the TFTP tree, defining a boot option for D-Installer + Iguana.
+2. Configure libvirt network to serve the tree.
+3. Prepare the initial ramdisk image (initrd), based on Iguana.
+4. Boot from PXE.
+
+## Set up the TFTP tree
+
+The TFTP files are available in the `tftpboot-installation-openSUSE-Tumbleweed-ARCH` packages. They
+are placed in `/usr/share/tftpboot-installation`, but you should copy them (or at least the `net`
+subdirectory) to customize them. In the example below we are just copying all of them.
+
+    zypper in tftpboot-installation-openSUSE-Tumbleweed-x86_64
+    mkdir /srv/tftpboot
+    cp -a /usr/share/tftpboot-installation/openSUSE-Tumbleweed-x86_64 /srv/tftpboot
```

Comment:
This works, but note that this is not needed.
Minimal setup is:
- installing `syslinux`
- copy `/usr/share/syslinux/pxelinux.0` from installed package to `/srv/tftpboot`
- create `/srv/tftpboot/pxelinux.cfg` directory
- create `/srv/tftpboot/pxelinux.cfg/default` as described below

syslinux is 2MB, tftpboot-installation is 500MB+

of course using syslinux will change paths in configs below, like not having `/srv/tftpboot/net` and `/srv/tftpboot/net/boot` and all files are related just to `/srv/tftpboot`

Reply:
> This is for custom modification of iguana, right?
> 
> Because for testing d-installer (without doing changes in iguana) it is enough to install `iguana` package which contains both kernel and initrd.
> 

We need to modify the initrd by now (to add the `/etc/NetworkManager` directory).

> Another option for tests is to use direct kernel boot instead of pxe booting. In virt-manager you can in boot loader section you can directly provide paths to kernel, initrd and set kernel options.

Yes, that's right. I personally have been using that approach until now, but it was a requirement to document a PXE setup for our QA team.